jwst/extract_2d/extract_2d.py

```diff
@@ -1,302 +1,16 @@
-    else:
-        # Get the value the user specified (if any)
-        if (user_type is not None) and (user_type in ['POINT', 'EXTENDED']):
-            # Use the value supplied by the user
-            log.info('Detected a {0} source type in slit {1}'.format(user_type, slit.name))
-            if user_type.strip().upper() == 'POINT':
-                result = True
-            else:
-                result = False
-        else:
-            log.info("Unknown source type")
-
-    return result
-
-
-def get_source_xpos(input_model, slit, slit_wcs, lam, msa_model):
-    """
-    Compute the source position within the slit for a NIRSPEC FS.
-
-    Parameters
-    ----------
-    input_model : `~jwst/datamodels/model_base.DataModel`
-        The input to ``extract_2d``.
-    slit : `~jwst/transforms/models/Slit`
-        The slit tuple.
-    slit_wcs : `~gwcs.wcs.WCS`
-        The WCS object for this slit.
-    lam : float
-        Wavelength in microns.
-
-    Returns
-    -------
-    xpos : float
-        X coordinate of the source as a fraction of the slit size.
-    """
-    xoffset = input_model.meta.dither.x_offset # in arcsec
-    yoffset = input_model.meta.dither.y_offset # in arcsec
-    v2ref = input_model.meta.wcsinfo.v2_ref # in arcsec
-    v3ref = input_model.meta.wcsinfo.v3_ref # in arcsec
-    v3idlyangle = input_model.meta.wcsinfo.v3yangle # in deg
-    vparity = input_model.meta.wcsinfo.vparity
-
-    idl2v23 = trmodels.IdealToV2V3(v3idlyangle, v2ref, v3ref, vparity)
-    # Compute the location in V2,V3 [in arcsec]
-    xv, yv = idl2v23(xoffset, yoffset)
-    # The NIRSPEC transforms expect V2,V3 positions in deg
-    xv = xv / 3600.
-    yv = yv / 3600.
-
-    v2v3_to_msa_frame = slit_wcs.get_transform("v2v3", "msa_frame")
-    xpos_abs, ypos_abs, lam = v2v3_to_msa_frame(xv, yv, lam)
-    xpos_frac = absolute2fractional(msa_model, slit, xpos_abs, ypos_abs)
-    return xpos_frac
-
-
-def get_msa_model(input_model):
-    # Get the reference file used in constructing the WCS.
-    msa_ref = input_model.meta.ref_file.msa.name
-    from .. import assign_wcs
-    from .. datamodels import MSAModel
-    step = assign_wcs.AssignWcsStep()
-    msa = MSAModel(step.reference_uri_to_cache_path(msa_ref))
-    return msa
-
-
-def absolute2fractional(msa_model, slit, xposabs, yposabs):
-    """
-    Compute the fractional position in ``x`` within the slit in MSA coordinates.
-
-    Parameters
-    ----------
-    input_model : `~jwst/datamodels/model_base.DataModel`
-        The input to ``extract_2d``.
-    slit : `~jwst/transforms/models/Slit`
-        The slit tuple.
-    xposabs, yposabs : float
-        (x, y) positions in the ``msa_frame``.
 
-    Returns
-    -------
-    xpos : float
-        The fractional X coordinates within the slit.
-    """
-    num, xcenter, ycenter, xsize, ysize = msa_model.Q5.data[slit.shutter_id]
-    return (xposabs - xcenter) / (xsize/ 2.)
+    if exp_type in nrs_modes:
```

Comment:
So what about using extract_2d for NRC_GRISM and NIS_WFSS? Is that part coming yet?

Reply:
...it's coming....it's coming....